test/test_ssltransport.py

```diff
@@ -91,11 +91,11 @@ def validate_response(
 
 def validate_peercert(ssl_socket: SSLTransport) -> None:
     binary_cert = ssl_socket.getpeercert(binary_form=True)
-    assert type(binary_cert) == bytes
+    assert isinstance(binary_cert, bytes)
```

Comment:
Doesn't this change the meaning of these asserts? Now subclasses would pass, too.

Reply:
It does, is this an issue?

Reply:
I'd prefer keeping as-is, does the tool say anything about why they're upgrading this? Very strange given that its a change in behavior.

Reply:
pycodestyle (flake8's dependency) had [a related change](https://pycodestyle.pycqa.org/en/latest/developer.html#id3:~:text=E721%3A%20adjust%20handling%20of%20type%20comparison.%20Allowed%20forms%20are%20now%20isinstance(x%2C%20t)%20or%20type(x)%20is%20t.%20PR%20%231086%2C%20%231167.) recently

The exact error is "E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`", so `assert type(binary_cert) is bytes` is allowed, this is a more correct way to check the type than with `==`

Reply:
Our tests are littered with isinstance() checks that could be more precise, but I revert the ones in this pull request (and a few more) in https://github.com/urllib3/urllib3/pull/3169/commits/b3fa87a944024023ae6f54f51b315fdca6ad19c9.